Sources/StreamChat/Models/Attachments/AttachmentTypes.swift

```diff
@@ -34,6 +34,25 @@ public enum LocalAttachmentState: Hashable {
     case uploadingFailed
     /// The attachment is successfully uploaded.
     case uploaded
+    /// The attachment is being downloaded.
+    case downloading(progress: Double)
+    /// The attachment download failed.
+    case downloadingFailed
+    /// The attachment has been downloaded.
+    case downloaded
+    
+    var isUploading: Bool {
+        switch self {
+        case .pendingUpload, .uploading, .uploadingFailed, .uploaded:
+            return true
+        case .unknown, .downloading, .downloadingFailed, .downloaded:
+            return false
+        }
+    }
+    
+    var isDownloading: Bool {
+        !isUploading
+    }
```

Comment:
This is kinda a breaking change since we are adding new cases, and our enums are all frozen. Could these be part of a different state maybe? Something like a new property `downloadingState`?

Reply:
Could be. Something like `LocalAttachmentDownloadState`.

Reply:
Done